### PR TITLE
The modified version with ARCHIBUS assignment

### PR DESCRIPTION
**Pull request by dingj2012**
This is the file upload application example using Spring MVC:
The source repository can be found on: 
    https://github.com/spring-guides/gs-uploading-files.git

### CODE REVIEW

**Comment by isobretatel** [comment]
How could your code be improved?

**Comment by dingj2012** [comment]
1. Add the option to store in either filesystem, server side or blob field.
2. Details add: keep comments complete and clean; consistent html interface.
3. Implement the persistence support

**Comment by isobretatel** [comment]
Besides adding new features, how could your code be improved?

**Comment by dingj2012** [comment]
1. The use of ServletContextAware supertype can be replaced by the API request.(for example, I could use like "request.getsession().getservletcontext()...."
2. The helper method should validate input. 
3. Rename class to FileController to avoid confussion.

**Comment by isobretatel** [comment]
Could you implement those improvements?
Could you also fix all warnings reported by Eclipse, CheckStyle, PMD, FindBugs?

**Comment by isobretatel** [comment]
Could your code be refactored to make it more readable and maintainable?

**Comment by dingj2012** [comment]
Update committed with refactoring and class replacement.

File: complete/src/main/java/hello/FileController.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	
+	return "redirect:upload";
+}
+
+
+
+//helper method to find the named file in the internal file system
+//return true if the file exists in the system; It validate the input first
+@Override
+private boolean checkFileinDir(String name,File file,
```
Comment (isobretatel) [inline complete/src/main/java/hello/FileController.java]: Separate business logic from UI.

File: complete/src/main/java/hello/FileController.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	return "redirect:upload";
+}
+
+
+
+//helper method to find the named file in the internal file system
+//return true if the file exists in the system; It validate the input first
+@Override
+private boolean checkFileinDir(String name,File file, 
+							RedirectAttributes redirectAttributes)
```
Comment (isobretatel) [inline complete/src/main/java/hello/FileController.java]: RedirectAttributes belongs to UI.

File: complete/src/main/java/hello/FileController.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+{
+	if (!name.isEmpty()){
+		try {
+			File[] list = file.listFiles();
```
Comment (isobretatel) [inline complete/src/main/java/hello/FileController.java]: File belongs to business logic.

File: complete/src/main/java/hello/FileController.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import org.springframework.ui.Model;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+public class FileController {
```
Comment (isobretatel) [inline complete/src/main/java/hello/FileController.java:21]: Refactor business logic into separate classes.

**Pull request closed without merging**